components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -170,6 +171,8 @@ func (r *WorkspaceReconciler) actOnStatus(ctx context.Context, workspace *worksp
 				}
 			}
 
+			r.deleteWorkspaceSecrets(ctx, workspace)
```

Comment:
lets check for an error, and not delete the workspace if there was an issue removing the secrets so we try again in the next reconcile tick

Reply:
Will add a retry for secret deletion in the token secret PR.